### ISSUE TITLE
Vuex module disappears

### ISSUE THREAD

**Issue by mjmnagy** [nuxt/nuxt.js#6711]
### Version

[v2.8.1](https://github.com/nuxt.js/releases/tag/v2.8.1)

### Reproduction link

[localhost:8000/norepo](localhost:8000/norepo)

### Steps to reproduce

I am trying to make a component library that will be shared across several apps. Each app is internal to a larger global app and has their own domain.

`hub.website.com`

`app.website.com`

The component library uses `Vuex` to allow users to change their preferences(Ex: want to increase the button size or change it's color).

In my components library project i created the `index.js` file which is the main file

    
    //index.js
    import * as vsComponents from './components'        // imports all the components
    import storeModule from './store/components'        //Imports the store module which is mentioned below

    const install = (Vue, options = {}) => {
      if (!options.store) {                             //Must pass the store from the larger component
        throw new Error('Please provide vuex store.')
      }

      //Register Store modules
      options.store.registerModule('components', storeModule) //Registers the module into the larger store

      // Use Components
      Object.values(vsComponents).forEach(vsComponent => { //Creates the components for vue to use
        Vue.use(vsComponent)
      })
    }

    if (typeof window !== 'undefined' && window.Vue) {    //No 100% sure i need this ??? - i would like it ssr so im thinking its not required (any help here would be awesome too)
      install(window.Vue)
    }

    export default install

    export * from './components/index'                 //Allows you to individually import each component

I use the following command to create the component library and upload it to bitbucket
`"scripts":{"production": "vue-cli-service build --target lib --name sc src/index.js"}` in which `src/index.js` is the file with the code pasted above.

In my nuxt application, i

 1. npm i bitbucket project
 2. Create a file `components.js` in the plugin folder
 3. Add the file to `nuxt.config.js` - i have played around with the ssr true/false

Here my `components.js` file

    import components from 'components'         //Component library
    import Vue from 'vue'

    export default context => {
      Vue.use(components, { store: context.store })
    }

**The Funny part**

On the initial load of the page, i can see the added state from the components library

   `<pre>`
     `{{ this.$store.state.components }} `
   `</pre>`

When i refresh the page, the added state is now gone?!?!?!

**TLDR**
Basically, i am importing a component lirbary as a plugin within nuxt. On the initial load, the plugin works as expect by injecting it's `vuex` module parameters into the `nuxt`'s store. Upon subsequent loads, the `vuex` module is gone???

Essentially, all im trying to do is have the a basic store module held within the components library. This would keep everything consistent between apps and then add the module to the larger app store and be able to utilize/use it. 

### What is expected ?

store module is registered and can be used across nuxt

### What is actually happening?

registers right after npm run dev and then disappears if the page is reloaded

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10043">#c10043</a>)</em></sub></div>

**Comment by aldarund** [comment]
Please provide a minimal working repo where your issue could be seen

**Comment by mjmnagy** [comment]
Unfortunately i do not have a repo
i believe it is due to persisted state issues

the options set in nuxt.config.js is setup when the server is initiated therefore i would need to ensure i install a plugin to keep the state on refresh and problem solved

**Comment by aldarund** [comment]
not quite follow about install plugin :)

**Comment by aldarund** [comment]
closing, if u still have this problem please open new issue with minimal reproduction repo
